From 04961d65f3f98f29f4a7cc007a1ae5bfc32e42ff Mon Sep 17 00:00:00 2001
From: Rafael Ostertag <rafisol@opencsw.org>
Date: Sat, 17 Sep 2011 12:39:57 +0200
Subject: [PATCH] Remove support for abstract sockets from
 g_socket_address_new_from_native

This makes gio/tests/unix-fd work.
---
 gio/gsocketaddress.c |    9 +++++++++
 1 files changed, 9 insertions(+), 0 deletions(-)

diff --git a/gio/gsocketaddress.c b/gio/gsocketaddress.c
index d4dcbb0..2482bd4 100644
--- a/gio/gsocketaddress.c
+++ b/gio/gsocketaddress.c
@@ -250,6 +250,13 @@ g_socket_address_new_from_native (gpointer native,
 #ifdef G_OS_UNIX
   if (family == AF_UNIX)
     {
+      /*
+       * Solaris doesn't support abstract sockets
+       */
+#ifdef __sun
+      return g_unix_socket_address_new_with_type ("", 0,
+						  G_UNIX_SOCKET_ADDRESS_ANONYMOUS);
+#else
       struct sockaddr_un *addr = (struct sockaddr_un *) native;
       gint path_len = len - G_STRUCT_OFFSET (struct sockaddr_un, sun_path);
 
@@ -275,6 +282,8 @@ g_socket_address_new_from_native (gpointer native,
 	}
       else
 	return g_unix_socket_address_new (addr->sun_path);
+#endif
+
     }
 #endif
 
-- 
1.7.6.1

